### ISSUE TITLE
Protontricks fails with warning: Unknown file arch of /home/$USER/.cache/protontricks/proton/GE-Proton9-1/bin/wineserver.

### ISSUE THREAD

**Issue by RaphaelDD885** [Matoking/protontricks#286]
**Describe the bug**
Protontricks fails with `warning: Unknown file arch of /home/$USER/.cache/protontricks/proton/GE-Proton9-1/bin/wineserver.`

**To Reproduce**
1. Run command `protontricks 268500 dotnet472`
2. Command fails and error is displayed

**Expected behavior**
Protontricks should install dotnet472. however this issue occurs with any game and command

**System (please complete the following information):**
 - Distro: Debian trixie/sid
 - Protontricks installation method: pipx
 - Protontricks version: 1.11.1
 - Steam version: beta

Output using -vv:
```
x86_64-linux-gnu-capsule-capture-libs: warning: we are assuming "_el_fn_sh_complete" to be private, but it's just a guess
i386-linux-gnu-capsule-capture-libs: warning: we are assuming "_el_fn_sh_complete" to be private, but it's just a guess
Executing cd /usr/bin
------------------------------------------------------
warning: Unknown file arch of /home/raphael/.cache/protontricks/proton/GE-Proton9-1/bin/wineserver.
------------------------------------------------------
```

**Comment by RaphaelDD885** [comment]
also occurs when running `protontricks --gui` or similar on any game

**Comment by Matoking** [comment]
The "unknown file arch" warning is an upstream issue with Winetricks: see the issue report at Winetricks/winetricks#2183. There are some proposed fixes in development, but none have been merged yet.

That said, you should be able to close the warnings and continue. Is there any output afterwards once you do so? What if you run `protontricks <APPID> -v`? This prints _a lot_ of output, so you'll want to use [gist.github.com](https://gist.github.com) to post the output.

**Comment by RaphaelDD885** [comment]
here you go
https://gist.github.com/RaphaelDD885/96eab29c728f8d33da9e30736443475b

**Comment by Matoking** [comment]
Figured out what's the problem: the Debian version of Winetricks includes an additional patch that's based on an old version of the pull request Winetricks/winetricks#2189. It's the `13_winetricks_get_file_arch__try_workaround_for_scripts_if_arch_is_unknown.patch` file included in the Debian source files [here](https://packages.debian.org/sid/winetricks).

This patch also changes the "Unknown file arch" message from a dismissable warning to an error that halts execution.

One option to workaround this might be to uninstall the distro version of Winetricks and install the upstream version using the instructions here:

https://github.com/Winetricks/winetricks?tab=readme-ov-file#manual-install

**Comment by RaphaelDD885** [comment]
thx, sadly i wont be able to check any results this week, since im not at home.

**Comment by CrispyHanako** [comment]
An addendum to my last post, I did a clean install of Arch (via archinstall) on a new machine with nothing on it except for the requirements to run protontricks, winetricks, steam, etc. All of which were installed via flatpaks when possible. I didn't get any errors this time but now I get stuck in a loop of "what do you want to do with this wineprefix" (install windows dll or component) and "what would you like to install". I select what I want to install and it starts the loop over again. I've legitimately clicked through this about 10 loops and it will not go away.

**Comment by Matoking** [comment]
The "unknown file arch" error should be fixed assuming you're running the development version of Protontricks *and* Winetricks.

The warning is only related to a check that Winetricks performs, and any other problems (eg. looping menus) are unrelated. The looping menu in particular is related to Winetricks and is probably out of scope for Protontricks.

**Comment by BarelyAPrincess** [comment]
> Hi! Ubuntu 24.04 Manual installation winetricks from https://github.com/Winetricks/winetricks?tab=readme-ov-file#manual-install Then install package **protontricks_1.12.0-1_all.deb**
> 
> ```
> protontricks --no-bwrap --gui
> Executing cd /usr/bin
> winetricks GUI enabled, using zenity 4.0.1
> ------------------------------------------------------
> warning: Unknown file arch of /home/$USER/.cache/protontricks/proton/Proton 9.0/bin/wineserver.
> ------------------------------------------------------
> ```

Simply following the steps to update from the manual-install guide was the exact trick I needed. I think it even fixed some issues I was having with other programs and didn't even realize they were the fault of Winetricks.

**Comment by emoxam** [comment]
So what is the fix?

**Comment by fwdekker** [comment]
Indeed, this appears to be an issue with (the Debian distribution of) Winetricks. Updating Winetricks to the latest development version resolves this issue entirely for me.

**Comment by Koli0842** [comment]
I have this on Arch, my winetricks is the last release on the upstream repo from January. Since protontricks releases are more frequent, I assume it is generally recommended to run off of winetricks-git instead?

**Comment by FriskTemmie** [comment]
> I have this on Arch, my winetricks is the last release on the upstream repo from January. Since protontricks releases are more frequent, I assume it is generally recommended to run off of winetricks-git instead?

using `sudo winetricks --self-update`, as ging-dev suggested fixed the issue on CachyOS for me, so it'll likely work on arch too. Just run that and then try to run protontricks again.

**Comment by Zefile8** [comment]
had the same error on mint and `sudo winetricks --self-update` worked for me as well.
